### PR TITLE
Feature/cbl heartbeat test

### PR DESCRIPTION
**Pull request by euniceatcouchbase**
#### Add CBL heartbeat feature testing

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- Add CBL heartbeat feature testing
- restructured the test suite to put heartbeat testing under topology test suite, since 3 machines are needed.

### CODE REVIEW

**Review by mgcouchbase — COMMENTED** [review]
Eunice, please add jenkins jobs

File: CBLClient/Replication.py [hunk 1]
```diff
@@ -29,14 +29,16 @@ def configure(self, source_db, target_url=None, target_db=None,
                   replication_type="push_pull", continuous=False,
                   push_filter=False, pull_filter=False, channels=None,
                   documentIDs=None, replicator_authenticator=None,
-                  headers=None, filter_callback_func='', conflict_resolver=""):
+                  headers=None, filter_callback_func='', conflict_resolver="", heartbeat=""):
```
Comment (mgcouchbase) [inline CBLClient/Replication.py:32]: could you please add single cotes for all

**Comment by euniceatcouchbase** [comment]
http://uberjenkins.sc.couchbase.com:8080/job/CBLITE_iOS-Listener-TestServer-Topology-specific-Functional-CBLITE-XATTRS-DELTA-SYNC-tests/1322/

File: libraries/provision/ansible/playbooks/install-nginx.yml [hunk 2]
```diff
@@ -5,6 +5,10 @@
   vars:
     upstream_sync_gatways:
     upstream_sync_gatways_admin:
+    keepalive_timeout: "keepalive_timeout 600s;"
+    proxy_send_timeout: "proxy_send_timeout 600s;"
+    proxy_read_timeout: "proxy_read_timeout 600s;"
```
Comment (sridevi-15) [inline libraries/provision/ansible/playbooks/install-nginx.yml:10]: this timeout won't effect SGW load balancer tests, right? If possible, can you run one single topology job with this branch?

**Review by sridevi-15 — COMMENTED** [review]
Can you please provide jenkins job with this test?
Also as there are changes on nginx which is used by SGW load balancer tests. Run one SGW topology job against this branch.
Otherwise I am okay with changes

File: libraries/provision/ansible/playbooks/install-nginx.yml (hunk 2 above)
Comment (euniceatcouchbase) [reply libraries/provision/ansible/playbooks/install-nginx.yml:10]: this is the original default value. if not requiring a customized timeout setting, default will be used.
could you please point me the SGW load balancer tests? upgrade job? I would like to run and verify before merge the code.

**Comment by euniceatcouchbase** [comment]
ran sync gateway load balancer jobs, ensure there is no regression caused by this change: http://uberjenkins.sc.couchbase.com:8080/job/cen7-sync-gateway-functional-tests-topology-xattrs-deltaSync-ServerSsl_MAGMA/228/testReport/

File: CBLClient/Replication.py (hunk 1 above)
Comment (euniceatcouchbase) [reply CBLClient/Replication.py]: updated in a new push

File: libraries/provision/ansible/playbooks/install-nginx.yml (hunk 2 above)
Comment (euniceatcouchbase) [reply libraries/provision/ansible/playbooks/install-nginx.yml:10]: load balancer job results link added

**Review by mgcouchbase — APPROVED** [review]

**Review by sridevi-15 — APPROVED** [review]

**Pull request merged**